### ISSUE TITLE
Unable to map number keys in table mode

### ISSUE THREAD

**Issue by mpenkov** [kaio/ibus-table#66]
I'm writing a custom table layout for Russian: https://github.com/mpenkov/ibus-russian-elecom/blob/master/russian-elecom.txt

So far, I've gotten everything to work, with the exception of the number keys and their shifted counterparts. I have this in my table:

```
### first row
!	!	1
"	"	1
#	№	1
$	;	1
%	%	1
&	:	1
'	?	1
(	*	1
)	(	1
~	)	1
=	=	1
~	~	1
|	|	1
```

So I expect # (shift + 3) on my keyboard to produce № when the layout is active. Unfortunately, pressing shift and 3 together produces #, ignoring the above mapping. I've confirmed that the layout is active (pressing other keys, e.g. qwerty outputs йцукен as expected).

I'd really appreciate it if you could look at the layout and let me know what I'm doing wrong.

Many thanks for your work, ibus is literally the only thing that satisfies my rather esoteric use case.

**Comment by mike-fabian** [comment]
I just tried your table and it worked well for me with ibus-table-1.12.4.
Typing a # (by typing Shift+3) does indeed produce a № for me.

**Comment by mike-fabian** [comment]
Your table contains:

```
### The below mapping is broken
\	\	1

...

### The below mapping is broken
\	/	1
...
```

That seems to work as well for me.

You map \ to two different characters, \ and /.

So you get a selection when you type \, like in the following screenshot:


![Screenshot](https://user-images.githubusercontent.com/2330175/110028444-8732f580-7d33-11eb-9f15-b5a9bc0f443b.png)

**Comment by mike-fabian** [comment]
Same with ~ which you map to two different characters:

![Screenshot](https://user-images.githubusercontent.com/2330175/110028699-dda03400-7d33-11eb-9f7f-01c241be0cd3.png)

**Comment by mike-fabian** [comment]
By the way, you can achieve the same result not only by writing a table for ibus-table, you can also write an input method for m17n-db and then use it with either ibus-m17n or ibus-typing-booster. 

Using it with ibus-m17n would achieve the basically the same as using ibus-table your russian-elecom table.

Using ibus-typing-booster might be better though because that can also give completions for words which you have typed previously.

If one does not want such completions, it is possible to setup ibus-typing-booster to emulate the simpler behaviour of ibus-m17n, like this:

http://mike-fabian.github.io/ibus-typing-booster/documentation.html#simulate-ibus-m17n

Writing an russian-elecom.mim input method for m17n-db is not any more difficult than writing a table for ibus-table,
actually I think it is even easier in this case.

An example of such a very simple m17n-db  input method is here:

https://github.com/mike-fabian/m17n-db-rovas

**Comment by mike-fabian** [comment]
Simple example:

```
;; russian-elecom.mim -- Russian Elecom input method

(input-method ru russian-elecom)

(description
"Russian Elecom input method")

(title "Russian Elecom")

(map
 (trans
  ("#" "№")
  ("Q" "Й")
  ("~" ")")  ; typing ~ once gives )
  ("~~" "~") ; typing ~ twice gives ~
  ;; Just more similar lines for the other stuff you need
  ))

(state
  (init
    (trans)))
```

**Comment by mike-fabian** [comment]
I just tested the partial file from my last comment by copying it to 

```
~/.m17n.d/russian-elecom.mim
```

then `ibus restart` and then `ibus list-engine | grep elecom` should show it:

```
$ ibus list-engine | grep elecom
  m17n:ru:russian-elecom - russian-elecom (m17n)
  table:russian-elecom - Russian (Elecom)
$
```

**Comment by mike-fabian** [comment]
Then one can add russian-elecom (m17n) in the Gnome control centre when using Gnome or using `ibus-setup` when using other desktops.

Or, one can add ibus-typing-booster in the Gnome control center or with ibus-setup and then in the setup tool of ibus-typing-booster add "russian-elecom".

**Comment by mike-fabian** [comment]
Screenshot of ibus-typing-booster setup to use russian-elecom.mim and a Russian hunspell dictionary.

I typed # to get the № then a Return and started the next line by typing Q which gave me the Й and a few possible completions from the Russian dictionary (the completions can be disabled in the setup tool if one does not want them):

![Screenshot](https://user-images.githubusercontent.com/2330175/110032883-f8c17280-7d38-11eb-8ce2-914fbeece3a7.png)

**Comment by mpenkov** [comment]
Thank you for your detailed attention to my problem.

I addressed the issues in my table layout, and regenerated the input method. I found that I had to rename the it (from russian-elecom to russian-ELECOM) for some of the changes to have effect. Something somewhere must be caching the old layout.

Anyway, this resolves the issue I was having. I will have a look at the alternative method you suggested at a later time.

Thanks again for your replies and your work on this project!
